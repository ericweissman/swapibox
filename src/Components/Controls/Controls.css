header {
  background-color: #24272b;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  color: yellow;
}
/*# sourceMappingURL=Controls.css.map */