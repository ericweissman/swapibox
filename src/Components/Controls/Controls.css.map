{
    "version": 3,
    "mappings": "AAAA,AAAA,MAAM,CAAC;EACL,gBAAgB,EAAE,OAAO;EACzB,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,MAAM;CACpB",
    "sources": [
        "Controls.scss"
    ],
    "names": [],
    "file": "Controls.css"
}