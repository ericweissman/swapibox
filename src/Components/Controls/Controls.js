import React, { Component } from 'react';
import '../../Main.scss';
import PropTypes from 'prop-types'
import { throws } from 'assert';
import { uid } from 'uid'


class Controls extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: ''
    }
  }

  handleClick = (event) => {
    const category = event.target.name
    event.preventDefault();
    this.props.populateData(category);
    this.props.updateActive(category);
    this.setState({active: category});
  }

  render() {
    const buttons = ['people', 'planets', 'vehicles', 'favorites']
    let { favorites } = this.props
    return(
      <nav>
        <h1>SWapiBox</h1>
        <div className="btn-container">
          {
            buttons.map((button) => {
              if (button !== 'favorites') {
                return (
                  <button className={this.state.active === button ? "btn active" : "btn"} name={button} onClick={this.handleClick}>{button}</button>
                )
              } else {
                return (
                  <button className={this.state.active === button ? "btn active" : "btn"} name={button} onClick={this.handleClick}>{button} {favorites.length}</button>
              )}
          })
        }
        </div>
      </nav>
    )
  }
}

Controls.propTypes = {
  populateData: PropTypes.func,
  updateActive: PropTypes.func
}

export default Controls;