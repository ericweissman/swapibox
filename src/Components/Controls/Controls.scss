nav {
  h1 {
    border: 2px solid $accent;
    padding: 10px;
    margin-left: 80px;
    border-radius: 20px;
    font-size: 50px;
  }
  background-color: $menus;
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: $accent;
  font-family: $h1-font;

  .btn {
    font-family: $title-font;
    text-transform: capitalize;
    cursor: pointer;
    outline: none;
    font-size: 22px;
    border-radius: 5px;
    padding: 5px;
    margin: 0px 20px;
    width: 150px;
    border: 2px solid $accent;
    background-color: $menus;
    color: $text;
    &:hover {
      background-color: $accent;
      color: black;
    }
  }
     .active {
      @extend .btn;
      background-color: $accent;
      color: black;
    }
  div {
    margin-right: 100px;
  }
  }
