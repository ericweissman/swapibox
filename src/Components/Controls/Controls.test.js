import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls'

describe('Controls', () => {
  const updateActiveMock = jest.fn();
  const populateDataMock = jest.fn();
  const mockFavorites = [1, 2];
  const expectedState = {active: ''}
  const mockEvent = { preventDefault: jest.fn(), target: {name: 'people'}}
  const mockEvent2 = { preventDefault: jest.fn(), target: {name: 'planets'}}
  let wrapper;
  beforeEach(() => {
     wrapper = shallow(<Controls updateActive={updateActiveMock} populateData={populateDataMock} favorites={mockFavorites} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()  
  });

  it('should have  the correct default state', () => {
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should call the the prop functions the correct times when handleClick is called', () => {
    wrapper.instance().handleClick(mockEvent)
    expect(updateActiveMock).toHaveBeenCalledTimes(1);
    expect(populateDataMock).toHaveBeenCalledTimes(1);
  })

  it('should call the prop functions with the correct params', () => {
    wrapper.instance().handleClick(mockEvent)
    expect(updateActiveMock).toHaveBeenCalledWith(mockEvent.target.name)
    expect(populateDataMock).toHaveBeenCalledWith(mockEvent.target.name)
  }) 

  it('should reset state when a button is clicked', () => {
    expect(wrapper.state()).toEqual(expectedState)
    wrapper.instance().handleClick(mockEvent)
    expect(wrapper.state()).toEqual({active: 'people'})
    wrapper.instance().handleClick(mockEvent2)
    expect(wrapper.state()).toEqual({active: 'planets'})
  })

})