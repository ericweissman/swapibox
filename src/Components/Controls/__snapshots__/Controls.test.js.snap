// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match the snapshot 1`] = `
<nav>
  <h1>
    SWapiBox
  </h1>
  <button
    className="btn"
    name="people"
    onClick={[Function]}
  >
    people
  </button>
  <button
    className="btn"
    name="planets"
    onClick={[Function]}
  >
    planets
  </button>
  <button
    className="btn"
    name="vehicles"
    onClick={[Function]}
  >
    vehicles
  </button>
  <button
    className="btn"
    name="favorites"
    onClick={[Function]}
  >
    favorites
     
    2
  </button>
</nav>
`;
